ReactSkia/views/common/RSkTextUtils.cpp

```diff
@@ -0,0 +1,89 @@
+ /*
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "ReactSkia/views/common/RSkTextUtils.h"
+
+using namespace skia::textlayout;
+
+namespace facebook {
+namespace react {
+namespace {
+inline int getLines(std::vector<LineMetrics>& metrics, float frameHeight) {
+    int i=0;
+    int minNumberOfLines = 0;
+    for(; i< metrics.size(); i++) {
+        if ((metrics[i]).fBaseline > frameHeight) {
+            minNumberOfLines = (metrics[i]).fLineNumber;
+            break;
+        }
+    }
+    return minNumberOfLines;
+}
+
+inline int getMaxNumberOfLines(std::vector<LineMetrics>& metrics, int maxNumberOfLines, float frameHeight) {
+    int minNumberOfLines = getLines(metrics, frameHeight);
+    if ((minNumberOfLines) && (minNumberOfLines < maxNumberOfLines))
+        return minNumberOfLines;
+        
+    return maxNumberOfLines;
+}
+
+inline SkScalar yPosOffset(AttributedString attributedString, SkScalar paraHeight, Float frameHeight ) {
+    for(auto &fragment: attributedString.getFragments()) {
+        if((paraHeight < frameHeight) && (!fragment.textAttributes.textAlignVertical.empty())) {
+            if(!strcmp(fragment.textAttributes.textAlignVertical.c_str(),"center"))
+                return ((frameHeight/2) - (paraHeight/2));
+            else if(!strcmp(fragment.textAttributes.textAlignVertical.c_str(),"bottom"))
+                return (frameHeight - paraHeight);
+        }
+    }
+    return 0;
+}
+} //namespace
+
+namespace RSkTextUtils{
+int getTextLines(float rectStrokeWidth, std::vector<LineMetrics> metrics, float textFrameWidth, float maxNumberofLines, float frameHeight) {
+    int numberOfLines = 0;
+    if (rectStrokeWidth)
+        numberOfLines = getLines(metrics, textFrameWidth);
+
+    if (maxNumberofLines) {
+        numberOfLines = getMaxNumberOfLines(metrics, maxNumberofLines, frameHeight);
+    }
+
+    return numberOfLines;
+}
+
+std::unique_ptr<skia::textlayout::Paragraph>& rebuildPara(ParagraphStyle paraStyle, std::shared_ptr<ParagraphBuilder> builder, float numberOfLines, float textFrameWidth, std::unique_ptr<skia::textlayout::Paragraph>& paragraph) {
+    paraStyle.setMaxLines(numberOfLines);
+    builder->setParagraphStyle(paraStyle);
+    if (numberOfLines) {
+        paragraph = builder->Build();
+        paragraph->layout(textFrameWidth);
+    }
+
+    return paragraph;
+}
+
+void drawTextBackground(SkCanvas *canvas, SharedColor backgroundColor, Rect frame, float isLineHeight) {
+    SkPaint paint;
+    if ((backgroundColor) || (isLineHeight >= 0)) {
```

Comment:
Do the LineHeight zero also demands drawing ?

Reply:
Do opacity needs to be considered to decide Draw needed or not. If it's fully transparent, still is it needed to be drawn ?

Reply:
Can't the draw background Api from drawutils be used Here ?

Reply:
For,
LineHeight =0 and opacity  =0,
We do not need to draw.